### ISSUE TITLE
Callback parameters for multiple types

### ISSUE THREAD

**Issue by rivo** [senchalabs/jsduck#643]
I often have a situation where two types are allowed for a parameter or config. If one of them is a function, I would like to be able to specify the function's signature but that currently does not work. Here's an example:

```js
/**
 * @cfg {number/Function} myCfg This can be either a number or a function.
 * When I specify the signature of the function, however, the result is not
 * as I expect.
 * @cfg {string} myCfg.in Some input string.
 * @cfg {string} myCfg.return Some return value.
 */
```

This turns into:

![image](https://cloud.githubusercontent.com/assets/480930/13827991/a176a276-ebbf-11e5-8e07-247d515cfc5b.png)

It works ok if the accepted type for this config is just `{Function}` but not if multiple types are allowed like `{number/Function}`. Is there a way to have jsduck recognize that what follows is the function's signature?

I understand that there may be an ambiguity if one specifies `{Object/Function}` but in that case, I would be ok with a fixed preference, e.g. additional subparameters would be interpreted as object fields.

**Comment by nene** [comment]
Yeah, it would make sense to detect if the type contains "Function" instead of checking that it strictly equals to "Function".

But truth to be told, the development of JSDuck has kinda stagnated. No idea when this might be implemented.

**Comment by rivo** [comment]
Out of curiosity, what's the reason for that stagnation?

**Comment by nene** [comment]
That's just me not being really motivated to develop it as I'm not developing with Sencha stack any more.

**Comment by rivo** [comment]
Ok, I understand. Actually, I'm not using it for a Sencha project. I like that JSDuck is so flexible that it will adapt to my own style and code organization while still generating very precise documentation.

I would probably contribute but I'm not very familiar with the stack (and I kind of skipped ruby) so there's a bit of a learning curve.

Anyway, it still works for now. It would be great if it could at least remain on life support. ;-)
